### PR TITLE
Fb photogram industrial 2

### PR DESCRIPTION
**Pull request by FadiBaker92**

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (APatel-AI) [inline app/controllers/comments_controller.rb:1]: Nice job on adding the notice messages. They create a great UI experience.

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#  index_comments_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (author_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: "User", counter_cache: true
+  belongs_to :photo
```
Comment (APatel-AI) [inline app/models/comment.rb]: I would recommend adding a `:counter_cache` for photos similar to how you added one for the author, to access the photos count.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#  index_likes_on_fan_id    (fan_id)
+#  index_likes_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User"
```
Comment (APatel-AI) [inline app/models/like.rb]: A `:counter_cache` would be really useful here to access the number of fans a user has.

File: app/models/like.rb
Comment (APatel-AI) [inline app/models/like.rb:1]: Add a validation for `:fan_id`. I would refer to the example given in Photogram Industrial Part 2 towards the end of the instructions.

File: app/models/photo.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Foreign Keys
+#
+#  fk_rails_...  (owner_id => users.id)
+#
+class Photo < ApplicationRecord
+  belongs_to :owner, class_name: "User", counter_cache: true 
+  has_many :comments
+  has_many :fans, through: :likes
+
```
Comment (APatel-AI) [inline app/models/photo.rb:27]: A photo also has many likes.

File: app/views/comments/show.json.jbuilder
Comment (APatel-AI) [inline app/views/comments/show.json.jbuilder:1]: Great job on the comment section files

File: app/views/likes/show.json.jbuilder
Comment (APatel-AI) [inline app/views/likes/show.json.jbuilder:1]: The likes section looks good

**Review by APatel-AI — CHANGES_REQUESTED** [review]
There are a couple of minor fixes needed.
Overall great job Fadi !! 👍.

File: app/views/likes/show.json.jbuilder
Comment (FadiBaker92) [reply app/views/likes/show.json.jbuilder:1]: thanks for feedback

File: app/views/comments/show.json.jbuilder
Comment (FadiBaker92) [reply app/views/comments/show.json.jbuilder:1]: thanks for feedback

File: app/models/photo.rb (hunk 3 above)
Comment (FadiBaker92) [reply app/models/photo.rb:27]: thanks for feedback

File: app/models/like.rb
Comment (FadiBaker92) [reply app/models/like.rb:1]: thanks for feedback.

File: app/models/like.rb (hunk 2 above)
Comment (FadiBaker92) [reply app/models/like.rb]: thanks for feedback

File: app/models/comment.rb (hunk 1 above)
Comment (FadiBaker92) [reply app/models/comment.rb]: thanks for feedback

File: app/models/comment.rb
Comment (APatel-AI) [inline app/models/comment.rb:1]: Looks good :) !! Thank you for adding the validation.

**Review by APatel-AI — APPROVED** [review]
Great job on the changes !!

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :own_photos, class_name: "Photo", foreign_key: "owner_id"  
+  has_many :comments, foreign_key: "author_id"
+  
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest" dependent: :destroy
```
Comment (thierrychau) [inline app/models/user.rb]: Missing a comma

**Review by thierrychau — CHANGES_REQUESTED** [review]
Great job! Thanks @APatel-AI  for the great feedback. I have one minor additional comment, see below.

File: app/models/user.rb (hunk 4 above)
Comment (FadiBaker92) [reply app/models/user.rb]: Thanks @thierrychau for the feedback! I went ahead and added them.

**Review by thierrychau — APPROVED** [review]
Good to merge, great job!

**Pull request merged**